src/Sfa.Das.Sas.Tools.MetaDataCreationTool/Services/AssessmentOrgsExcelPackageService.cs

```diff
@@ -38,12 +38,26 @@ public IEnumerable<Organisation> GetAssessmentOrganisations(ExcelPackage package
                             Street = organisationsWorkSheet.Cells[i, 7].Value != null ? organisationsWorkSheet.Cells[i, 7].Value.ToString() : string.Empty,
                             Town = organisationsWorkSheet.Cells[i, 8].Value != null ? organisationsWorkSheet.Cells[i, 8].Value.ToString() : string.Empty,
                             Postcode = organisationsWorkSheet.Cells[i, 9].Value != null ? organisationsWorkSheet.Cells[i, 9].Value.ToString() : string.Empty
-                        }
-                    };
+                        },
+                        Ukprn = CheckForValidUkprn(organisationsWorkSheet.Cells[i, 10].Value?.ToString())
+                };
                 }
             }
         }
 
+        public int? CheckForValidUkprn(string ukprnDetails)
+        {
+            int? ukprnProcessed = null;
```

Comment:
Should this not check against a regex instead? 
```
const string ukprnRegex = @"^\d{8}$";
ukprnCellValue = ukprnCellValue.Trim();
bool isValid = Regex.IsMatch(ukprnCellValue, ukprnRegex);
```

Not sure you need to check if the Provider is active also?